### PR TITLE
Defining a method to allow diagm(V) for a vector V 

### PR DESCRIPTION
**Pull request by karthikrangasai**
Defining a missing method to allow diagm(V) for a vector V rather diagm(0=>v)

### CODE REVIEW

File: stdlib/LinearAlgebra/src/dense.jl [hunk 1]
```diff
@@ -271,6 +271,26 @@ julia> diagm(1 => [1,2,3], -1 => [4,5])
...
+        end
+    end
+    return A
+end
+function diagm_container(kv::AbstractVector...)
+    T = promote_type(map(x -> eltype(x), kv)...)
+    n = mapreduce(x -> length(x), max, kv)
+    return zeros(T, n, n)
+end
+function diagm_container(kv::BitVector...)
```
Comment (StefanKarpinski) [inline stdlib/LinearAlgebra/src/dense.jl]: The above method does the exact same thing, doesn't it?

**Comment by StefanKarpinski** [comment]
So if I'm understanding this correctly, it appears to take zero or more diagonal vectors and add them to the diagonal? That's somewhat fancy but I'm not sure that we really need more than just `diagm(v)`.

**Comment by fredrikekre** [comment]
See #31125.

File: stdlib/LinearAlgebra/src/dense.jl (hunk 1 above)
Comment (karthikrangasai) [reply stdlib/LinearAlgebra/src/dense.jl]: Yeah, I just duplicated the function such that it does it only for a k value of zero and any number of vectors

**Pull request closed without merging**